### PR TITLE
chore(ci): better snyk error COMPASS-7693

### PR DESCRIPTION
**Pull request by mcasimir**

### CODE REVIEW

**Review by himanshusinghs — APPROVED** [review]

**Review by lerouxb — APPROVED** [review]

File: scripts/snyk-test.js [hunk 1]
```diff
@@ -26,7 +26,10 @@ async function snykTest(cwd) {
           '--dev',
           `--json-file-output=${tmpPath}`,
         ],
-        { cwd }
+        {
+          cwd,
+          maxBuffer: 1024 * 1024 * 5, // 5 MB (default is 1)
```
Comment (addaleax) [inline scripts/snyk-test.js:31]: [nit] We could as well raise this to be 50 MB :) Or use `spawn` with I/O passed through instead of `execFile`

**Review by addaleax — APPROVED** [review]

File: scripts/snyk-test.js (hunk 1 above)
Comment (mcasimir) [reply scripts/snyk-test.js:44]: i thought about both: 
- using spawn and forwarding stdio, we silenced the output recently cause it kept confusing folks about what are the actual vulnerabilities that needs attention
- 50 MB that we can do yeah, any reason why we may want to have a smaller limit instead?

Comment (addaleax) [reply scripts/snyk-test.js:44]: > any reason why we may want to have a smaller limit instead?

Not really, I don't think 50 MB is something where we'd need to worry about memory exhaustion or anything like that

File: scripts/snyk-test.js [hunk 2]
```diff
@@ -6,12 +6,25 @@ const { glob } = require('glob');
...
+}
+
 async function snykTest(cwd) {
   const tmpPath = path.join(os.tmpdir(), 'tempfile-' + Date.now());
 
   let execErr;
 
   try {
+    if (!(await fileExists(path.join(cwd, `package.json`)))) {
+      return;
```
Comment (lerouxb) [inline scripts/snyk-test.js:25]: You're pushing `null` onto results again.

Comment (mcasimir) [reply scripts/snyk-test.js:25]: nope, that's ok, the `"null"` issue, only happens with the monorepo root, but in the monorepo root a package.json is granted to be there, cause that package.json is being read to fetch workspaces. The issue is not in this function in any case.

Comment (mcasimir) [reply scripts/snyk-test.js:25]: https://github.com/mongodb-js/compass/blob/594d5049270abd08869276da331732010b792afd/scripts/snyk-test.js#L81

**Pull request merged**